freqtrade/plugins/pairlist/MarketCapPairList.py

```diff
@@ -0,0 +1,92 @@
+                'for "pairlist.config.number_assets"')
+
+        self._stake_currency = config['stake_currency']
+        self._number_pairs = self._pairlistconfig['number_assets']
+        self._refresh_period = self._pairlistconfig.get('refresh_period', 6*60*60)
+        self._pair_cache: TTLCache = TTLCache(maxsize=1, ttl=self._refresh_period)
+        self._url = \
+            f'https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page={self._number_pairs}&page=1&sparkline=false'   # noqa
+        if not self._exchange.exchange_has('fetchTickers'):
+            raise OperationalException(
+                'Exchange does not support dynamic whitelist. '
+                'Please edit your config and restart the bot.'
+            )
+
+    @property
+    def needstickers(self) -> bool:
+        """
+        Boolean property defining if tickers are necessary.
+        If no Pairlist requires tickers, an empty Dict is passed
+        as tickers argument to filter_pairlist
+        """
+        return True
+
+    def short_desc(self) -> str:
+        """
+        Short whitelist method description - used for startup-messages
+        """
+        return f"{self.name} - top {self._pairlistconfig['number_assets']} marketcap pairs."
+
+    def gen_pairlist(self, tickers: Dict) -> List[str]:
+        """
+        Generate the pairlist
+        :param tickers: Tickers (from exchange.get_tickers()). May be cached.
+        :return: List of pairs
+        """
+        # Generate dynamic whitelist
+        # Must always run if this pairlist is not the first in the list.
+        pairlist = self._pair_cache.get('pairlist')
+        if not pairlist:
+            resp = requests.get(self._url)
+            data = resp.json()
+            pairlist = [entry['symbol'].upper() + f'/{self._stake_currency}' for entry in data]
+            pairlist = self.filter_pairlist(pairlist, tickers)
+            self._pair_cache['pairlist'] = pairlist
+
+        return pairlist
+
+    def filter_pairlist(self, pairlist: List[str], tickers: Dict) -> List[str]:
+        """
+        Filters and sorts pairlist and returns the whitelist again.
+        Called on each bot iteration - please use internal caching if necessary
+        :param pairlist: pairlist to filter or sort
+        :param tickers: Tickers (from exchange.get_tickers()). May be cached.
+        :return: new whitelist
+        """
+        # Use the incoming pairlist.
+        pairlist = [k for k in tickers.keys() if k in pairlist]
+        # Validate whitelist to only have active market pairs
+        pairlist = self._whitelist_for_active_markets(pairlist)
+        pairlist = self.verify_blacklist(pairlist, logger.info)
+        # Limit pairlist to the requested number of pairs
+        pairlist = pairlist[:self._number_pairs]
+        self.log_once(f"Searching {self._number_pairs} pairs: {pairlist}", logger.info)
+
+        return pairlist
```

Comment:
I don't think this does what it advertises.

While the generator (only called if this is the first pairlist) uses marketcap to sort the pairlist - this does not sort by this, but  simply returns the already existing pairlist without sorting it by marketcap (which is what i'd expect / what the documentation entry states).

Reply:
Good catch, the pairlist was sorted actually, but I'm iterating trough the tickerlist. I've fixed that so its useful until the other things are fixed...

Reply:
Also fixed.